From df64ae5f2782d1f998cb737c1ceb9cfd8ae472d6 Mon Sep 17 00:00:00 2001
From: Xu Wang <vulab@iscas.ac.cn>
Date: Thu, 2 Dec 2021 07:55:35 +0000
Subject: [PATCH backport 5.10 47/73] mctp: Remove redundant if statements

The 'if (dev)' statement already move into dev_{put , hold}, so remove
redundant if statements.

Signed-off-by: Xu Wang <vulab@iscas.ac.cn>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit d9e56d1839fa40dbaab640ec205390826bddf8ae)
---
 net/mctp/route.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/mctp/route.c b/net/mctp/route.c
index 046a5013c589..978c945ccea5 100644
--- a/net/mctp/route.c
+++ b/net/mctp/route.c
@@ -889,8 +889,7 @@ int mctp_local_output(struct sock *sk, struct mctp_route *rt,
 	if (!ext_rt)
 		mctp_route_release(rt);
 
-	if (dev)
-		dev_put(dev);
+	dev_put(dev);
 
 	return rc;
 
-- 
2.20.1

